From e034b2cc9d2125f80dabb698271a346cbcff8d76
From: Maciej W. Rozycki <macro@codesourcery.com>
Date: Tue, 5 Aug 2014 16:17:15 +0100
Subject: [PATCH] MIPS: Fix a .pdr section linker buffer overrun

	* elfxx-mips.c (_bfd_mips_elf_discard_info): Set section's
	rawsize if changing size.
---
 bfd/ChangeLog    |    5 +++++
 bfd/elfxx-mips.c |    2 ++
 2 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/bfd/ChangeLog b/bfd/ChangeLog
--- a/bfd/ChangeLog
+++ b/bfd/ChangeLog
@@ -1,3 +1,8 @@
+2014-08-05  Maciej W. Rozycki  <macro@codesourcery.com>
+
+	* elfxx-mips.c (_bfd_mips_elf_discard_info): Set section's
+	rawsize if changing size.
+
 2014-05-28  Tristan Gingold  <gingold@adacore.com>
 
 	* dwarf2.c (lookup_address_in_function_table): Add best_fit_len
diff --git a/bfd/elfxx-mips.c b/bfd/elfxx-mips.c
--- a/bfd/elfxx-mips.c
+++ b/bfd/elfxx-mips.c
@@ -12210,6 +12210,8 @@ _bfd_mips_elf_discard_info (bfd *abfd, struct elf_reloc_cookie *cookie,
   if (skip != 0)
     {
       mips_elf_section_data (o)->u.tdata = tdata;
+      if (o->rawsize == 0)
+	o->rawsize = o->size;
       o->size -= skip * PDR_SIZE;
       ret = TRUE;
     }
-- 
